base/shell/cmd/dir.c

```diff
@@ -973,13 +976,15 @@ DirPrintWideList(PDIRFINDINFO ptrFiles[],       /* [IN] Files' Info */
         if (ptrFiles[i]->stFindInfo.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)
         {
             /* Directories need 2 additional characters for brackets */
-            if ((_tcslen(ptrFiles[i]->stFindInfo.cFileName) + 2) > iLongestName)
-                iLongestName = _tcslen(ptrFiles[i]->stFindInfo.cFileName) + 2;
+            cxWidth = ConGetTextWidth(ptrFiles[i]->stFindInfo.cFileName) + 2;
+            if (cxWidth > iLongestName)
+                iLongestName = cxWidth;
```

Comment:
```suggestion
```

Reply:
Looks like this repeated code can be moved outside this if-else block (see below).